pkgs/by-name/qw/qwen-code-bin/update.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p curl jq nix-prefetch-url
+
+set -euo pipefail
+
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+NIX_FILE="package.nix"
+GITHUB_REPO="QwenLM/qwen-code"
+ASSET_NAME="gemini.js"
+REV_PREFIX="v"
+
+CURRENT_VER=$(grep -oP 'version = "\K[^"]+' "${NIX_FILE}" || { echo "Error: Failed to extract version" >&2; exit 1; })
+
+API_URL="https://api.github.com/repos/${GITHUB_REPO}/releases/latest"
+RELEASE_JSON=$(curl --fail -s "${API_URL}" || { echo "Error: Failed to fetch GitHub release info" >&2; exit 1; })
```

Comment:
This must optionally take a GITHUB_TOKEN env to avoid rate limits. Please see other update scripts for examples.